Error while checking build/O2/fullCI for 9e467d10be13c48b71497cfbf94319354836e2b8 at 2023-08-22 19:02:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2Physics-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/11804/9e467d10be13c48b71497cfbf94319354836e2b8/build_O2_fullCI/pretty.html).

--- a/Common/Utils/src/fpu.cxx
+++ b/Common/Utils/src/fpu.cxx
@@ -33,8 +33,10 @@ static void __attribute__((constructor)) trapfpe()
 {
   // allows to disable set of particular FE's by setting corresponding bit of the O2_DISABLE_FPE_TRAP,
   // i.e. to enable only FE_DIVBYZERO use O2_DISABLE_FPE_TRAP=9
-  const char* ev = std::getenv("O2_DISABLE_FPE_TRAP");
-  int enabledFE = (FE_INVALID | FE_DIVBYZERO | FE_OVERFLOW) & ~(ev ? atoi(ev) : 0);
+  //  const char* ev = std::getenv("O2_DISABLE_FPE_TRAP");
+  //  int enabledFE = (FE_INVALID | FE_DIVBYZERO | FE_OVERFLOW) & ~(ev ? atoi(ev) : 0);
+  const char* ev = std::getenv("O2_ENABLE_FPE_TRAP");
+  int enabledFE = (FE_INVALID | FE_DIVBYZERO | FE_OVERFLOW) & (ev ? atoi(ev) : 0);
   if (enabledFE) {
     feenableexcept(enabledFE);
   }

--- a/Common/Utils/src/fpu.cxx
+++ b/Common/Utils/src/fpu.cxx
@@ -1,0 +1,42 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+// To enable check for problematic LIBXML version uncomment this
+// #define _PROTECT_LIBXML_
+
+#ifdef __linux
+#ifdef _PROTECT_LIBXML_
+#include <libxml/xmlversion.h>
+#if LIBXML_VERSION > 20912
+#define _DUMMY_FEE_TRAP_
+#endif // LIBXML_VERSION > 20912
+#endif // _PROTECT_LIBXML_
+#else  // __linux
+#define _DUMMY_FEE_TRAP_
+#endif // __linux
+
+#ifdef _DUMMY_FEE_TRAP_
+void trapfpe() {}
+#else // _DUMMY_FEE_TRAP_
+#define _GNU_SOURCE 1
+#include <cfenv>
+#include <cstdlib>
+static void __attribute__((constructor)) trapfpe()
+{
+  // allows to disable set of particular FE's by setting corresponding bit of the O2_DISABLE_FPE_TRAP,
+  // i.e. to enable only FE_DIVBYZERO use O2_DISABLE_FPE_TRAP=9
+  const char* ev = std::getenv("O2_DISABLE_FPE_TRAP");
+  int enabledFE = (FE_INVALID | FE_DIVBYZERO | FE_OVERFLOW) & ~(ev ? atoi(ev) : 0);
+  if (enabledFE) {
+    feenableexcept(enabledFE);
+  }
+}
+#endif // _DUMMY_FEE_TRAP_

--- a/Utilities/rANS/src/FrequencyTable.cxx
+++ b/Utilities/rANS/src/FrequencyTable.cxx
@@ -71,7 +71,8 @@ inline double_t computeEntropy(const FrequencyTable& table)
 
 count_t computeRenormingPrecision(const FrequencyTable& frequencyTable)
 {
-  const uint8_t minBits = std::ceil(std::log2(frequencyTable.getNUsedAlphabetSymbols()));
+  const auto nused = frequencyTable.getNUsedAlphabetSymbols();
+  const uint8_t minBits = nused > 0 ? std::ceil(std::log2(nused)) : 0;
   const uint8_t estimate = minBits * 3u / 2u;
   const uint8_t maxThreshold = std::max(minBits, MaxRenormThreshold);
   const uint8_t minThreshold = std::max(estimate, MinRenormThreshold);

--- a/cmake/O2AddExecutable.cmake
+++ b/cmake/O2AddExecutable.cmake
@@ -92,7 +92,7 @@ function(o2_add_executable baseTargetName)
   endif()
 
   # add the executable with its sources
-  add_executable(${target} ${A_SOURCES})
+  add_executable(${target} ${A_SOURCES} ${CMAKE_SOURCE_DIR}/Common/Utils/src/fpu.cxx)
 
   # set the executable output name
   set_property(TARGET ${target} PROPERTY OUTPUT_NAME ${exeName})
@@ -115,6 +115,9 @@ function(o2_add_executable baseTargetName)
     endif()
     target_link_libraries(${target} PUBLIC ${lib})
   endforeach()
+
+  # needed for fpu.c
+  target_link_libraries(${target} PUBLIC ROOT::XMLIO)
 
   if(NOT A_NO_INSTALL)
     # install the executable

